pkg/test/framework/components/istio/operator.go

```diff
@@ -168,11 +168,31 @@ func deploy(ctx resource.Context, env *kube.Environment, cfg Config) (Instance,
 	}
 
 	// Deploy the Istio control plane(s)
+	ingressIPs := make(map[resource.ClusterIndex]string)
 	for _, cluster := range env.KubeClusters {
 		if cluster.IsControlPlaneCluster() {
 			if err := deployControlPlane(i, cfg, cluster, iopFile); err != nil {
 				return nil, fmt.Errorf("failed deploying control plane to cluster %d: %v", cluster.Index(), err)
 			}
+			igSvc, err := cluster.GetService("istio-system", "istio-ingressgateway")
+			if err != nil {
+				return nil, fmt.Errorf("failed getting ingressgateway service for cluster %d: %v", cluster.Index(), err)
+			}
+			if len(igSvc.Status.LoadBalancer.Ingress) < 1 {
+				return nil, fmt.Errorf("istio-ingressgateway service has no LoadBalancer ingress for cluster %d", cluster.Index())
```

Comment:
I am not sure this will work as expected. Typically the lb takes a few minutes to get provisioned, so there is probably a race between deployControlPlane and here. More importantly, in CI we run https://kind.sigs.k8s.io/ which has no lb. Instead it uses nodePort. this is confgured by the (poorly named) flag ` --istio.test.kube.minikube`. You can see some examples of getting the ip at https://github.com/istio/istio/blob/fe3615bf2e6d625842f66308d880310f432d1daf/pkg/test/framework/components/ingress/kube.go#L60 (possibly this could be reused a bit)

Reply:
Gotcha. If we use NodePort in this case, we don't currently have a way to tell remote clusters to use a different port for the remote pilot, right? 

https://github.com/istio/istio/blob/master/manifests/charts/base/templates/endpoints.yaml#L26

Reply:
Trying to work around this with a manual `apply` but Envoys in the remote cluster still can't connect to istiod. 
```
echo '
apiVersion: v1
kind: Endpoints
metadata:
  name: istiod-remote
  namespace: istio-system
subsets:
- addresses:
  - ip: 172.17.0.2
  ports:
  - port: 32535
    name: tcp-istiod
---
apiVersion: v1
kind: Service
metadata:
  name: istiod-remote
  namespace: istio-system
spec:
  ports:
  - port: 15012
    targetPort: 32535 
    name: tcp-istiod
  clusterIP: None
' | k --context=$CLUSTER1_CTX -n istio-system apply -f -
```